pkg/cache/errors/errors.go

```diff
@@ -0,0 +1,20 @@
+package errors
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ErrResourceNotCached indicates that the resource type
```

Comment:
I know I told you something else over Slack, but I realized there is another case of a cache error which is not in a subpackage: https://github.com/kubernetes-sigs/controller-runtime/blob/b1d6919d3e12fa85a119dd9792bdfdc17bdf8c3b/pkg/cache/informer_cache.go#L42

Lets maybe move this into the cache package, too?